smack-extensions/src/main/java/org/jivesoftware/smackx/usertune/UserTuneManager.java

```diff
@@ -0,0 +1,146 @@
+ * <br>
+ * To add a UserTune listener in order to remain updated with other users UserTune, use {@link #addUserTuneListener(UserTuneListener)} method.
+ * <br>
+ * To link a UserTuneElement with {@link Message}, use 'message.addExtension(userTuneElement)'.
+ * <br>
+ * An example to illustrate is provided inside UserTuneElementTest inside the test package.
+ * <br>
+ * @see <a href="https://xmpp.org/extensions/xep-0118.html">
+ *     XEP-0118: User Tune</a>
+ */
+public final class UserTuneManager extends Manager {
+
+    public static final String USERTUNE_NODE = "http://jabber.org/protocol/tune";
+    public static final String USERTUNE_NOTIFY = USERTUNE_NODE + "+notify";
+
+    private static final Map<XMPPConnection, UserTuneManager> INSTANCES = new WeakHashMap<>();
+
+    private final Set<UserTuneListener> userTuneListeners = new HashSet<>();
+    private final AsyncButOrdered<BareJid> asyncButOrdered = new AsyncButOrdered<>();
+    private final PubSubManager pubSubManager;
+    private boolean ENABLE_USER_TUNE_NOTIFICATIONS_BY_DEFAULT = true;
+
+    public static synchronized UserTuneManager getInstanceFor(XMPPConnection connection) throws NotLoggedInException {
+        UserTuneManager manager = INSTANCES.get(connection);
+        if (manager == null) {
+            manager = new UserTuneManager(connection);
+            INSTANCES.put(connection, manager);
+        }
+        return manager;
+    }
+
+    private UserTuneManager(XMPPConnection connection) throws NotLoggedInException {
+        super(connection);
+        PepManager.getInstanceFor(connection).addPepListener(new PepListener() {
+            @Override
+            public void eventReceived(EntityBareJid from, EventElement event, Message message) {
+                if (!USERTUNE_NODE.equals(event.getEvent().getNode())) {
+                    return;
+                }
+
+                final BareJid contact = from.asBareJid();
+                asyncButOrdered.performAsyncButOrdered(contact, () -> {
+                    ItemsExtension items = (ItemsExtension) event.getExtensions().get(0);
+                    PayloadItem<?> payload = (PayloadItem) items.getItems().get(0);
+                    UserTuneElement tune = (UserTuneElement) payload.getPayload();
+
+                    for (UserTuneListener listener : userTuneListeners) {
+                        synchronized (userTuneListeners) {
+                            listener.onUserTuneUpdated(contact, message, tune);
+                        }
+                    }
+                });
+            }
+        });
+        if (ENABLE_USER_TUNE_NOTIFICATIONS_BY_DEFAULT) {
+            enableUserTuneNotifications();
+        }
+        pubSubManager = PubSubManager.getInstanceFor(getAuthenticatedConnectionOrThrow(), connection().getUser().asBareJid());
+    }
+
+    public void setDefaultNotificationsValue(boolean bool) {
```

Comment:
The method name is not clear. Should probably be `setUserTuneNotificationsEnabledByDefault(boolean)`